File: 프로그래머스/2/42586. 기능개발/기능개발.py
from math import ceil

def solution(progresses, speeds):
    curr_day = 0
    done = 0
    answer = []
    
    for p, s in zip(progresses, speeds):
        need_day = ceil((100 - p) / s)
        if curr_day < need_day:
            answer.append(done)
            curr_day = need_day
            done = 0
        done += 1
    
    answer.append(done)
    return answer[1:]

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.ArrayDeque;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        ArrayDeque<Integer> deque = new ArrayDeque<>();
        int currentDays = 0;
        int done = 0;
        
        for(int i = 0; i < progresses.length; i++){
            int needDays = (int) Math.ceil((100.0 - progresses[i]) / speeds[i]);
            if(needDays > currentDays){
                deque.addLast(done);
                done = 0;
                currentDays = needDays;
            }
            done++;
        }
        
        deque.addLast(done);
        deque.removeFirst();
        return deque.stream().mapToInt(i->i).toArray();
        
    }
}